### DISCUSSION TITLE
Add "Tabs" to tables with a lot of data columns

### DISCUSSION THREAD

**Discussion by jcc5018** [filamentphp/filament#11686] [Ideas]
The only tabs functionality I see in the docs seem to be to create quick filters of data. 

This is not what i need. 

I have a few models in which i have a ton of columns. Lets just say 100 for sake of illustration:

obviously viewing 100 data points would require scrolling. 

But I was hoping that there was a way to  do something like this:

 ```
public static function table(Table $table): Table
    {
        return $table
            ->columns([
      
                TextColumn::make('name')
                          ->searchable()
                          ->sortable(),
                 TextColumn::make('column2')
                          ->searchable()
                          ->sortable(),
         TABGROUP::make('group_1")
                      ->columns([
                          ...6 more columns
                          ]),
         TABGROUP::make('group_2")
                      ->columns([
                          ...6 more columns
                          ])  ,
                          
                          Actions::   ...

..etc.

```

In this set up, the first 2 columns would display on every tab selection such as a frozen column in EXCEL
but each tab would display additional columns while maintaining the sort order or pagination of the given model

trailing columns such as actions could also be added as long as it is outside the TABGROUP

so in short, all table columns are the same but instead of one huge table where one might need to scroll, columns within TABGROUPS can be switched out for a different set. Any filters applied will search the entire table as if all data columns were displayed. 

the Tab group only shows or hides the given set of columns.

**Comment by AxelvdS** [comment]
Hi, I was looking for something similar but couldn't find a solution with tabs. Instead I have implemented an action that does the same. For me it was just a need to show cost columns or not. The action does that nicely and is easy to replicate so you could have 10 actions for different sets of columns.
